app/src/test/java/org/simple/clinic/onboarding/OnboardingEffectHandlerTest.kt

```diff
@@ -0,0 +1,49 @@
+package org.simple.clinic.onboarding
+
+import com.f2prateek.rx.preferences2.Preference
+import com.nhaarman.mockitokotlin2.eq
+import com.nhaarman.mockitokotlin2.mock
+import com.nhaarman.mockitokotlin2.verify
+import com.nhaarman.mockitokotlin2.verifyNoMoreInteractions
+import org.junit.After
+import org.junit.Test
+import org.simple.clinic.mobius.EffectHandlerTestCase
+
+class OnboardingEffectHandlerTest {
+
+  private val uiActions = mock<OnboardingUi>()
+  private val hasUserCompletedOnboarding = mock<Preference<Boolean>>()
+
+  private val effectHandler = OnboardingEffectHandler(
+      hasUserCompletedOnboarding = hasUserCompletedOnboarding,
+      viewEffectsConsumer = OnboardingViewEffectHandler(uiActions)::handle
+  ).build()
+  private val testCase = EffectHandlerTestCase(effectHandler)
+
+  @After
+  fun tearDown() {
+    testCase.dispose()
+  }
+
+  @Test
+  fun `when complete onboarding effect is received, then mark user has completed onboarding`() {
```

Comment:
👍🏻  This is great. Thought about this: we should follow TDD while adding these tests. Like, write the test then see if it's working with the effect handler previously and then check if it's working with the view effect handler. Saying this because once I had mistakenly added the wrong UI action both in the test and in the view effect. 😅

Reply:
Yeah, it's a good idea. Currently this is moving existing controller/logic tests into the new tests, so all of the effect handler and update functions code is already present. So, we can move these tests directly.